src/api-serverless/src/proxies/proxies.routes.ts

```diff
@@ -0,0 +1,219 @@
+import { Request, Response } from 'express';
+import { asyncRouter } from '../async.router';
+import { getWalletOrThrow, needsAuthenticatedUser } from '../auth/auth';
+import { ApiResponse } from '../api-response';
+import { CreateNewProfileProxy } from '../generated/models/CreateNewProfileProxy';
+import * as Joi from 'joi';
+import { getValidatedByJoiOrThrow } from '../validation';
+import { profilesService } from '../../../profiles/profiles.service';
+import { profileProxyApiService } from './proxy.api.service';
+import { ProfileProxyEntity } from '../../../entities/IProfileProxy';
+import { BadRequestException } from '../../../exceptions';
+import { ProxyApiRequestAction } from './proxies.api.types';
+import { CreateNewProfileProxyAllocateRepAction } from '../generated/models/CreateNewProfileProxyAllocateRepAction';
+import { CreateNewProfileProxyAllocateCicAction } from '../generated/models/CreateNewProfileProxyAllocateCicAction';
+import { CreateNewProfileProxyCreateWaveAction } from '../generated/models/CreateNewProfileProxyCreateWaveAction';
+import { CreateNewProfileProxyReadWaveAction } from '../generated/models/CreateNewProfileProxyReadWaveAction';
+import { CreateNewProfileProxyActionType } from '../generated/models/CreateNewProfileProxyActionType';
+import { CreateNewProfileProxyRateWaveDropAction } from '../generated/models/CreateNewProfileProxyRateWaveDropAction';
+import { assertUnreachable } from '../../../helpers';
+import { ProfileProxyActionEntity } from '../../../entities/IProfileProxyAction';
+import { profileProxyActionApiService } from './proxy-action.api.service';
+
+const router = asyncRouter();
+
+router.post(
+  '/',
+  needsAuthenticatedUser(),
+  async (
+    req: Request<any, any, CreateNewProfileProxy, any, any>,
+    res: Response<ApiResponse<ProfileProxyEntity>>
+  ) => {
+    const { body } = req;
+    const newProxy = getValidatedByJoiOrThrow(body, NewProfileProxySchema);
+    const grantorProfile = await profilesService
+      .getProfileAndConsolidationsByHandleOrEnsOrIdOrWalletAddress(
+        getWalletOrThrow(req)
+      )
+      ?.then((result) => result?.profile ?? null);
+    if (!grantorProfile) {
+      throw new BadRequestException(
+        'You need to create a profile before you can create a proxy'
+      );
+    }
+
+    if (
+      grantorProfile.external_id.toLowerCase() ===
+      newProxy.target_id.toLowerCase()
+    ) {
+      throw new BadRequestException('You cannot create a proxy to yourself');
+    }
+
+    const profileProxy = await profileProxyApiService.createProfileProxy({
+      params: newProxy,
+      grantorProfile
```

Comment:
input structure is strange. shouldn't the whole argument be params with structure:

```
{
  grantor_id: string;
  target_id: string;
}
```